src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -707,6 +707,8 @@ void SolverInterfaceImpl::setMeshTriangle(
 {
   PRECICE_TRACE(meshID, firstEdgeID,
                 secondEdgeID, thirdEdgeID);
+  PRECICE_CHECK(_dimensions == 3, "setMeshTriangle is only possible for 3D cases."\
```

Comment:
Is it necessary to add an assertion check for dim != 3 when we have a check?

Reply:
The assertions we need in both `mesh` functions. To be sure that those are not used the wrong way in the future by somebody else.